packages/core/src/sync-historical/index.ts

```diff
@@ -242,7 +242,7 @@ export const createHistoricalSync = async (
         undefined
       ) {
         throw new Error(
-          `Detected inconsistent RPC responses. Log with transaction hash ${log.transactionHash} not found in block ${block.hash}.`,
+          `Detected inconsistent eth_getLogs response. 'log.transactionHash' ${log.transactionHash} not found in 'block.transactions' ${block.hash}`,
```

Comment:
The reason I called in "inconsistent RPC response" instead of "inconsistent eth_getLogs response" is that it's entirely possible that eth_getBlockByNumber is the method returning the bad data.